app/controllers/SocialAuthController.scala

```diff
@@ -33,7 +34,7 @@ class SocialAuthController @Inject() (
             _ <- authInfoRepository.save(profile.loginInfo, authInfo)
             authenticator <- silhouette.env.authenticatorService.create(profile.loginInfo)
             value <- silhouette.env.authenticatorService.init(authenticator)
-            result <- silhouette.env.authenticatorService.embed(value, Redirect(routes.HomeController.google))
+            result <- silhouette.env.authenticatorService.embed(value, Redirect("http://localhost:3000"))
```

Comment:
Temporary hardcode.. need to figure out way to preserve a query param after redirecting out to / back from the auth provider. Since we are handling auth via silhouette, seems like we actually have to hit this endpoint directly (not AJAX)? But afterwards we'll need to point back to client URL, which should come from the initial client request. Let me know if that doesn't make sense or if you have other thoughts.

Reply:
https://www.silhouette.rocks/docs/social-state